src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -1185,42 +1150,105 @@ static void res_recv_cb(grpc_exec_ctx *exec_ctx, void *arg, grpc_error *error) {
     gpr_log(GPR_ERROR, "Invalid LB response received: '%s'",
             gpr_dump_slice(response_slice, GPR_DUMP_ASCII));
     gpr_slice_unref(response_slice);
-
-    /* Disconnect from server returning invalid response. */
-    op->op = GRPC_OP_SEND_CLOSE_FROM_CLIENT;
-    op->flags = 0;
-    op->reserved = NULL;
-    op++;
-    grpc_call_error call_error = grpc_call_start_batch_and_execute(
-        exec_ctx, lb_client->lb_call, ops, (size_t)(op - ops),
-        &lb_client->close_sent);
-    GPR_ASSERT(GRPC_CALL_OK == call_error);
+    grpc_call_cancel(glb_policy->lb_call, NULL);
```

Comment:
Actually, I don't think we want to cancel/close. That'd only result in a retry and reconnection to the same channel, which, by the pick first policy, would likely end up at the same server. I've changed to code to log and ignore. Do you think more drastic measures are necessary? Such as eliminating the server that returned the invalid serverlist and recreating the channel.